galite-core/src/main/kotlin/org/kopi/galite/form/VFixnumField.kt

```diff
@@ -33,7 +830,49 @@ class VFixnumField {
      * @param     maxVal          the maximal value the fixnum field can get.
      */
     fun computeWidth(digits: Int, scale: Int, minVal: Fixed?, maxVal: Fixed?): Int {
-      TODO()
+      var upperBound: Fixed
+      var lowerBound: Fixed
+      upperBound = calculateUpperBound(digits, scale)
+      lowerBound = upperBound.negate()
+      if (minVal != null && minVal > lowerBound) {
+        lowerBound = minVal.setScale(scale)
+      }
+      if (maxVal != null && maxVal < upperBound) {
+        upperBound = maxVal.setScale(scale)
+      }
+      return Math.max(upperBound.toString().length, lowerBound.toString().length)
     }
+
+    /**
+     * Computes the number of digits of a fixed field : FIXED(width, scale)
+     *
+     * @param     width           the width of the fixed field.
+     * @param     scale           the number of digits representing the fractional part.
+     */
+    fun computeDigits(width: Int, scale: Int): Int {
+      return if (scale == 0) {
+        width - width / 4
+      } else if (width == scale || width == scale + 1) {
+        scale
+      } else {
+        // decimal = width - scale - 1
+        // digits = decimal - decimal/4 + scale
+        width - 1 - ((width - scale - 1) / 4)
+      }
+    }
+  }
+
+  /**
+   * Constructor
+   */
+  init {
+    fieldMaxScale = maxScale
+    this.maxScale = maxScale
+    minval = minval?.setScale(fieldMaxScale) ?: calculateUpperBound(digits, fieldMaxScale).negate()
+    maxval = maxval?.setScale(fieldMaxScale) ?: calculateUpperBound(digits, fieldMaxScale)
+    this.digits = digits
+    this.fraction = fraction
```

Comment:
We dont need to add this.property = arg when the property is already declared as val or var in the constructor